server/src/main/java/org/elasticsearch/search/aggregations/bucket/global/GlobalAggregatorFactory.java

```diff
@@ -42,13 +43,13 @@ public GlobalAggregatorFactory(String name,
     @Override
     public Aggregator createInternal(SearchContext searchContext,
                                         Aggregator parent,
-                                        boolean collectsFromSingleBucket,
+                                        CardinalityUpperBound cardinality,
                                         Map<String, Object> metadata) throws IOException {
         if (parent != null) {
             throw new AggregationExecutionException("Aggregation [" + parent.name() + "] cannot have a global " + "sub-aggregation [" + name
                     + "]. Global aggregations can only be defined as top level aggregations");
         }
-        if (collectsFromSingleBucket == false) {
+        if (cardinality != CardinalityUpperBound.ONE) {
```

Comment:
nit: I know it was like this when you found it, but since you're touching this anyway, maybe add a message to the exception?